src/remount-fs/remount-fs.c

```diff
@@ -69,11 +71,96 @@ static int do_remount(const char *path, bool force_rw, Hashmap **pids) {
         return track_pid(pids, path, pid);
 }
 
+static int device_node_uuid(const char *path, sd_id128_t *ret_uuid) {
+        _cleanup_(sd_device_unrefp) sd_device *device = NULL;
+        int r;
+
+        r = sd_device_new_from_devname(&device, path);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to create device from %s: %m", path);
+
+        const char *uuid;
+        r = sd_device_get_property_value(device, "ID_FS_UUID", &uuid);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to get filesystem UUID for device %s: %m", path);
+
+        r = sd_id128_from_string(uuid, ret_uuid);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to parse ID_FS_UUID '%s' for device %s: %m",
+                                         uuid, path);
```

Comment:
one of those days we should introduce device_get_property_id128() as a helper, the same way we already have device_get_property_bool() and device_get_property_int()